src/test/article.spec.js

```diff
@@ -304,3 +311,246 @@ describe('GET api/v1/articles', () => {
+      await generateReport({
+        reporterId: reporter.id,
+        articleId: createArticle.id,
+        violation: 'Discrimination',
+        description: 'description'
+      })
+    );
+    const { description } = createdReport[0].dataValues;
+    const token = helper.jwtSigner({ id: reporter.id });
+    const res = await server()
+      .post(`/api/v1/articles/${createdArticle2.id}/report`)
+      .set('token', token)
+      .send({ description });
+    expect(res.status).to.equal(400);
+    expect(res.body.message).to.equal('violation is required');
+  });
+
+  it('should return validation error if description is  not provided', async () => {
+    const reporter = await createUser();
+    const author = await createUser();
+    const createdArticle = await createArticle(
+      await generateArticle({ authorId: author.id })
+    );
+
+    const createdArticle2 = await createArticle(
+      { id: 2000 },
+      await generateArticle({ authorId: author.id })
+    );
+    const createdReport = await reportArticle(
+      createdArticle.id,
+      await generateReport({
+        reporterId: reporter.id,
+        articleId: createArticle.id,
+        violation: 'Discrimination'
+      })
+    );
+    const { violation } = createdReport[0].dataValues;
+    const token = helper.jwtSigner({ id: reporter.id });
+    const res = await server()
+      .post(`/api/v1/articles/${createdArticle2.id}/report`)
+      .set('token', token)
+      .send({ violation });
+    expect(res.status).to.equal(400);
+    expect(res.body.message).to.equal('description is required');
+  });
+
+  it('should return validation error if description character is less than 10', async () => {
+    const reporter = await createUser();
+    const author = await createUser();
+    const createdArticle = await createArticle(
+      await generateArticle({ authorId: author.id })
+    );
+    const createdReport = await reportArticle(
+      createdArticle.id,
+      await generateReport({
+        reporterId: reporter.id,
+        articleId: createArticle.id,
+        violation: 'Discrimination',
+        description: 'some'
+      })
+    );
+    const { violation, description } = createdReport[0].dataValues;
+
+    const token = helper.jwtSigner({ id: reporter.id });
+    const res = await server()
+      .post(`/api/v1/articles/${createdArticle.id}/report`)
+      .set('token', token)
+      .send({ violation, description });
+    expect(res.status).to.equal(400);
+    expect(res.body.message).to.equal(
+      'description length must be at least 10 characters long'
+    );
+  });
+});
+describe('POST api/v1/articles/:articleId/report', () => {
+  beforeEach(async () => {
+    await db.User.destroy({ cascade: true, truncate: true });
+    await db.Article.destroy({ cascade: true, truncate: true });
+    await db.Report.destroy({ cascade: true, truncate: true });
+  });
+  it('should return a validation error if the article was already reported', async () => {
```

Comment:
I don't think this should be true.

A user should be able to report an article multiple times.